plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -775,6 +775,17 @@ public Optional<Object> getInfo(ConnectorTableHandle tableHandle)
     @Override
     public List<SchemaTableName> listTables(ConnectorSession session, Optional<String> optionalSchemaName)
     {
+        if (optionalSchemaName.isEmpty()) {
+            Optional<List<SchemaTableName>> allTables = metastore.getAllTables();
```

Comment:
if `optionalSchemaName` points to a single schema then `metastore.getAllTables` does more work than required, which can lead to slower perf compared to older impl.

cc: @kokosing @huberty89

Reply:
> if optionalSchemaName points to a single schema

but then we don't enter this code path, do we?

Reply:
indeed, i should read more carefully. Such an embarrassing mistake. 😄